lucille-core/src/main/java/com/kmwllc/lucille/core/RunDetails.java

```diff
@@ -0,0 +1,176 @@
+package com.kmwllc.lucille.core;
+
+import java.time.Instant;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import com.kmwllc.lucille.core.Runner.RunType;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+@Schema(
+    description = "Details about a specific run, including metadata, start time, and processing stats.")
+public class RunDetails {
+
+  @Schema(description = "Unique identifier for the run.", example = "run123")
+  private final String runId;
+
+  @Schema(description = "Configuration ID for the run.")
+  private String configId;
+
+  @Schema(description = "Start time of the run in ISO-8601 format.",
+      example = "2023-12-20T14:48:00.000Z")
+  private final Instant startTime;
+
+  @Schema(
+      description = "End time of the run in ISO-8601 format. Null if the run has not completed.",
+      example = "2023-12-20T15:48:00.000Z")
+  private Instant endTime;
+
+  @Schema(description = "Number of documents processed during the run.", example = "5000")
+  private long docsProcessed;
+
+  @Schema(description = "Number of errors encountered during the run.", example = "5")
+  private long errorCount;
+
+  @Schema(hidden = true)
+  private final transient CompletableFuture<Void> future;
+
+  @Schema(description = "Run result")
+  private RunResult runResult;
+
+  @Schema(description = "Current status")
+  private volatile String status = "new";
+
+  @Schema(description = "Run type")
+  private RunType runType;
+
+  @Schema(description = "Exception of run")
+  private Throwable exception;
+
+
+  public RunDetails(String runId, String configId) {
+    this.runId = runId;
+    this.configId = configId;
+    this.future = new CompletableFuture<>();
```

Comment:
It seems like "future" is being set to an empty CompletableFuture that never gets populated
This "future" is different from the one RunnerManager creates in runWithConfig where it does `CompletableFuture.runAsync(() -> {`. RunnerManager used to save that future in an instance variable but it's not doing that anymore. So it seems like the actual future that's being used to contain the Lucille run is never stored anywhere but instead we're storing an empty future. How are tests passing then? Please do let me know if I'm overlooking something.

Reply:
Important catch, thanks, looking into it.

Reply:
If you are only trying to store the status of the run here in RunDetails, please do not use a dummy CompletableFuture, that is not what it's meant for. It's meant for executing asynchronous tasks but you are not using it to execute anything here. If you want to be able to block in a thread-safe way you would need to have the "real" CompleteableFuture that contains the lucille run and you'd call get() or join() on it. If you do not need to do that here but only want to set/and get status information about the run, consider a simple boolean isDone or simply check whether endTime has been set.

Reply:
updated